SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -192,11 +192,15 @@ bool CNEMOEulerVariable::SetPrimVar(unsigned long iPoint, CFluidModel *FluidMode
   nonPhys = Cons2PrimVar(Solution[iPoint], Primitive[iPoint],
                          dPdU[iPoint], dTdU[iPoint], dTvedU[iPoint], eves[iPoint], Cvves[iPoint]);
 
+  /*--- Reset solution to previous one, if nonphys ---*/
   if (nonPhys) {
     for (iVar = 0; iVar < nVar; iVar++)
       Solution(iPoint,iVar) = Solution_Old(iPoint,iVar);
   }
 
+  /*--- Set additional point quantaties ---*/
+  Gamma(iPoint) = fluidmodel->ComputeGamma();  
+

```

Comment:
why here? I know it doesn't make a difference in the end, but just for tydiness, shouldn't this be computed next to other fluidmodel quantities, i.e, in cons2primvar ?

Reply:
Yeah, I was running into issues trying to do this.  So I followed the strategy in the NS version of this.

Since Cons2PrimVar doesnt have access to iPoint, I can't directly set it.  The other option would be to update Cons2PrimVar to return gamma.   Unless Im missing something.

Reply:
ok I see the problem. then I think it's ok like this. Otherwise, something we could do, or not, just for tidyness, is gather all these variables that are not primitive or conservative (gamma, dPdstuff, ccves, eves) in some sort of Secondary or someething vector, and pass it the same way we do with primitives and conservatives, instead of passing/computing one by one... depends on how much longer if you feel lile post-poning your real-work you were trying to avoid xD

Reply:
Ill make it a TODO in the NEMO project! hahah.  I think this will be the last general PR.   I need to start doing the Implicit!